### PR TITLE
Added Soviet02a

### PR DESCRIPTION
**Pull request by abcdefg30**
Supersedes #8279.

Open points:
- [x] https://github.com/OpenRA/OpenRA/pull/8285#discussion-diff-31284635
- [x] https://github.com/OpenRA/OpenRA/pull/8285#discussion-diff-31284750
- [x] https://github.com/OpenRA/OpenRA/pull/8285#discussion-diff-31284831

### CODE REVIEW

File: mods/ra/maps/soviet-02a/soviet02a.lua [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+		-- Destroy the bridge
+		if not bridgepart1.IsDead then
+			bridgepart1.Kill()
+		end
+		if not bridgepart2.IsDead then
+			bridgepart2.Kill()
+		end
+
+		-- Move units towards the oilpump
+		--[[Utils.Do(FleeingUnits, function(unit)
```
Comment (abcdefg30) [inline mods/ra/maps/soviet-02a/soviet02a.lua]: I commented this out as I don't get the reason for this.
What should I do with it?

File: mods/ra/maps/soviet-02a/soviet02a.lua [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+	-- Four infantry from the small island move towards the USSR command center and attack it after 24 Seconds
+	Trigger.AfterDelay(DateTime.Seconds(24),  function()
+		Utils.Do(CmdAtk, function(unit)
+			unit.AttackMove(AttackWaypoint1.Location)
+			Trigger.OnIdle(unit, unit.Hunt)
+			--Trigger.OnIdle(unit, function() unit.AttackMove(AttackWaypoint1.Location) end)
+		end)
+
+		-- On reaching waypoint 7 remove the trigger to keep moving or substitue with new instruction
+		--[[Trigger.OnEnteredFootprint({ AttackWaypoint1.Location }, function(unit, id)
```
Comment (abcdefg30) [inline mods/ra/maps/soviet-02a/soviet02a.lua]: Similar case here. I cut this down to [two lines](https://github.com/OpenRA/OpenRA/pull/8285/files#diff-b79eef01e62d56a3c8ca44d3db6e74efR113).

File: mods/ra/maps/soviet-02a/soviet02a.lua [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+	-- When destroying the allied radar dome or the refinery drop 2 badgers with 5 grenadiers each
+	Trigger.OnAnyKilled({ AlliedDome, AlliedProc }, function()
+		local powerproxy = Actor.Create("powerproxy.paratroopers", true, { Owner = player })
+		powerproxy.SendParatroopers(ParadropLZ.CenterPosition, false, Facing.South)
+		powerproxy.SendParatroopers(ParadropLZ.CenterPosition, false, Facing.SouthEast)
+		powerproxy.Destroy()
+	end)
+
+	-- When passing the map coordinates in CameraTriggerArea reveal the allied base
+	--[[Trigger.OnEnteredFootprint(CameraTriggerArea, function(unit, id)
```
Comment (abcdefg30) [inline mods/ra/maps/soviet-02a/soviet02a.lua]: I don't think a camera is necessary, as the player probably has the entire base already revealed using yaks.

File: OpenRA.Mods.Common/Scripting/Properties/PlaneProperties.cs [hunk 4]
```diff
@@ -35,17 +35,4 @@ public void ReturnToBase(Actor airfield = null)
...
 	}
-
-	[ScriptPropertyGroup("Combat")]
-	public class PlaneCombatProperties : ScriptActorProperties, Requires<AttackPlaneInfo>
-	{
-		public PlaneCombatProperties(ScriptContext context, Actor self)
-			: base(context, self) { }
-
-		[Desc("Fly an attack against the target actor.")]
-		public void Attack(Actor target)
```
Comment (Mailaender) [inline OpenRA.Mods.Common/Scripting/Properties/PlaneProperties.cs]: Why are you removing public API here?!?

Comment (Mailaender) [inline OpenRA.Mods.Common/Scripting/Properties/PlaneProperties.cs]: Nevermind.

**Comment by Mailaender** [comment]
The generic Attack still works for aircraft. Tested Intervention mission. :white_check_mark:

File: mods/ra/maps/soviet-02a/map.yaml [hunk 5]
```diff
@@ -0,0 +1,699 @@
...
+		Playable: True
+		AllowBots: False
+		Required: True
+		LockRace: True
+		Race: soviet
+		LockColor: True
+		ColorRamp: 3,255,127
+		LockSpawn: True
+		LockTeam: True
+		Enemies: Germany, Neutral
```
Comment (obrakmann) [inline mods/ra/maps/soviet-02a/map.yaml]: Remove Neutral (and, if needed, add a separate faction instead)

File: mods/ra/maps/soviet-02a/map.yaml [hunk 6]
```diff
@@ -0,0 +1,699 @@
...
+			ShowOwnerRow: false
+	^Vehicle:
+		Tooltip:
+			GenericVisibility: Enemy
+			ShowOwnerRow: false
+	^Plane:
+		Tooltip:
+			GenericVisibility: Enemy
+			ShowOwnerRow: false
+	^Wall:
```
Comment (obrakmann) [inline mods/ra/maps/soviet-02a/map.yaml]: `^Husk` is missing

File: mods/ra/maps/soviet-02a/soviet02a.lua [hunk 7]
```diff
@@ -0,0 +1,170 @@
+
```
Comment (obrakmann) [inline mods/ra/maps/soviet-02a/soviet02a.lua]: Spurious newline

File: mods/ra/maps/soviet-02a/soviet02a.lua [hunk 8]
```diff
@@ -0,0 +1,170 @@
...
+CameraTriggerArea = { CPos.New(42, 45), CPos.New(43, 45), CPos.New(44, 45), CPos.New(45, 45), CPos.New(46, 45), CPos.New(47, 45), CPos.New(48, 45), CPos.New(48, 56), CPos.New(48, 57), CPos.New(48, 58), CPos.New(48, 59), CPos.New(40, 63), CPos.New(41, 63), CPos.New(42, 63), CPos.New(43, 63), CPos.New(44, 63), CPos.New(45, 63), CPos.New(46, 63), CPos.New(47, 63) }
+PassingBridgeLocation = { CPos.New(59, 56), CPos.New(60, 56) }
+
+CmdAtk = { Attacker1, Attacker2, Attacker3, Attacker4 }
+FleeingUnits = { Fleeing1, Fleeing2 }
+HuntingUnits = { Hunter1, Hunter2, Hunter3, Hunter4 }
+
+
```
Comment (obrakmann) [inline mods/ra/maps/soviet-02a/soviet02a.lua]: Spurious newline

File: mods/ra/maps/soviet-02a/soviet02a.lua [hunk 9]
```diff
@@ -0,0 +1,170 @@
...
+	end)
+	Trigger.OnObjectiveCompleted(player, function(p, id)
+		Media.DisplayMessage(p.GetObjectiveDescription(id), "Objective completed")
+	end)
+	Trigger.OnObjectiveFailed(player, function(p, id)
+		Media.DisplayMessage(p.GetObjectiveDescription(id), "Objective failed")
+	end)
+
+	CommandCenterIntact = player.AddPrimaryObjective("Protect the Command Center.")
+	DestroyallAllied = player.AddPrimaryObjective("Destroy all Allied units and structures.")
```
Comment (obrakmann) [inline mods/ra/maps/soviet-02a/soviet02a.lua]: `DestroyAllAllied`

File: mods/ra/maps/soviet-02a/soviet02a.lua [hunk 10]
```diff
@@ -0,0 +1,170 @@
...
+		player.MarkFailedObjective(CommandCenterIntact)
+	end)
+
+	Trigger.AfterDelay(0, function()
+		local buildings = Map.ActorsInBox(Map.TopLeft, Map.BottomRight, function(self) return self.Owner == germany and self.HasProperty("StartBuildingRepairs") end)
+		Utils.Do(buildings, function(actor)
+			Trigger.OnDamaged(actor, function(building, attacker)
+				if building.Owner == germany and building.Health < building.MaxHealth * 0.8 then
+					building.StartBuildingRepairs()
+					if attacker.Type ~= "Yak" and not AlreadyHunting then
```
Comment (obrakmann) [inline mods/ra/maps/soviet-02a/soviet02a.lua]: not sure if the case is important here. Should probably be `yak`.

**Comment by obrakmann** [comment]
Another thing I noticed is that both harvesters get stuck after the first field is harvested, ie. they don't find the ore to the south. I guess on this small map we can just increase the search radius to cover the whole map.

**Comment by abcdefg30** [comment]
Updated: Fixed all comments and added a search radius of 50 for the harvester.

**Comment by abcdefg30** [comment]
"Fixed" the open points as suggested by @Mailaender.

**Comment by Mailaender** [comment]
Mission works as promised. :white_check_mark:

**Comment by Mailaender** [comment]
Nice work as always.

[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/0893cfc36d6baec165ccac37acb1342750b74578...964442505583b4e0e94d8f47e17690960c45b787)

**Pull request merged**